### PR TITLE
MGMT-3333: Show CVO in the UI

### PR DESCRIPTION
**Pull request by irosenzw**
Installing the cluster - Installing state:
installing-state

Installing the cluster - Finalizing state (web console URL is available) :
finalizing state

Popover:
final 2

CVO messages after 'View Cluster Events' link has been clicked:
CVO messages on cluster events

Installation complete:
install complete

 

Signed-off-by: Ido Rosenzwig <irosenzw@redhat.com>

### CODE REVIEW

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 1]
```diff
@@ -62,6 +82,107 @@ const getInstallationStatus = (cluster: Cluster) => {
...
+type FinalizingProgressProps = {
+  cluster: Cluster;
+};
+
+const FinalizingProgress: React.FC<FinalizingProgressProps> = ({ cluster }) => {
+  const { status } = cluster;
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+  const closeModal = () => setIsModalOpen(false);
+  return (
+    <div>
```
Comment (mareklibra) [inline src/components/clusterDetail/ClusterProgress.tsx:141]: nit: just fragment `<>` would be enough

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 2]
```diff
@@ -62,6 +82,107 @@ const getInstallationStatus = (cluster: Cluster) => {
...
+        entityKind="cluster"
+      />
+      <Flex className="pf-u-mr-3xl" display={{ default: 'inlineFlex' }}>
+        <FlexItem>{getFinalizingStatusIcon(status)}</FlexItem>
+        <FlexItem>
+          {status === 'finalizing' ? (
+            <Popover
+              headerContent={<div>Finalizing</div>}
+              bodyContent={
+                <div>
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx]:
```suggestion
                <TextContent>
```

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 3]
```diff
@@ -62,6 +82,107 @@ const getInstallationStatus = (cluster: Cluster) => {
...
+                  <div>
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:159]:
```suggestion
                  <Text>
```

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 4]
```diff
@@ -62,6 +82,108 @@ const getInstallationStatus = (cluster: Cluster) => {
...
+      />
+      <Flex className="pf-u-mr-3xl" display={{ default: 'inlineFlex' }}>
+        <FlexItem>{getFinalizingStatusIcon(status)}</FlexItem>
+        <FlexItem>
+          {status === 'finalizing' ? (
+            <Popover
+              zIndex={300} // set the zIndex below Cluster Events Modal
+              headerContent={<div>Finalizing</div>}
+              bodyContent={
+                <>
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:158]:
```suggestion
                <TextContent>
```

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 5]
```diff
@@ -62,6 +82,108 @@ const getInstallationStatus = (cluster: Cluster) => {
...
+                  <div>
+                    Finalizing the installtion, this may take a while. For more information see
+                    Cluster Events
+                  </div>
+                  <br />
+                  <Button
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:164]: The button can be put into Popover footer

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 6]
```diff
@@ -70,6 +192,7 @@ const ClusterProgress: React.FC<ClusterProgressProps> = ({ cluster }) => {
   const { status, hosts } = cluster;
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:192]:
```suggestion
  const { status, hosts = [] } = cluster;
```

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 7]
```diff
@@ -93,6 +216,19 @@ const ClusterProgress: React.FC<ClusterProgressProps> = ({ cluster }) => {
         variant={getProgressVariant(status)}
         className="cluster-progress-bar"
       />
+      <Flex className="pf-u-mt-md" display={{ default: 'inlineFlex' }}>
+        <FlexItem>
+          <HostProgress hosts={cluster.hosts as Host[]} hostRole="master" />
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:221]:
```suggestion
          <HostProgress hosts={hosts} hostRole="master" />
```

File: src/components/clusterDetail/ClusterProgress.tsx [hunk 8]
```diff
@@ -93,6 +216,19 @@ const ClusterProgress: React.FC<ClusterProgressProps> = ({ cluster }) => {
...
+        </FlexItem>
+        {isWorkersPresent && (
+          <FlexItem>
+            <HostProgress hosts={cluster.hosts as Host[]} hostRole="worker" />
```
Comment (jtomasek) [inline src/components/clusterDetail/ClusterProgress.tsx:225]:
```suggestion
            <HostProgress hosts={hosts} hostRole="worker" />
```

**Review by jtomasek — CHANGES_REQUESTED** [review]

**Review by jtomasek — CHANGES_REQUESTED** [review]

**Comment by andybraren** [comment]
Screenshots please 🙏

**Comment by nirfarkas** [comment]
@irosenzw I’ve got feedback on the “Pending” icon and changed it in the [mock](https://marvelapp.com/prototype/f2j8hh7/screen/72564353)

**Review by jtomasek — APPROVED** [review]

**Pull request merged**